tests/pfcwd/test_pfcwd_timer_accuracy.py

```diff
@@ -194,35 +196,11 @@ def verify_pfcwd_timers(self):
         """
         Compare the timestamps obtained and verify the timer accuracy
         """
-        self.all_detect_time.sort()
-        self.all_restore_time.sort()
-        logger.info("Verify that real detection time is not greater than configured")
-        config_detect_time = self.timers['pfc_wd_detect_time'] + self.timers['pfc_wd_poll_time']
-        err_msg = ("Real detection time is greater than configured: Real detect time: {} "
-                   "Expected: {} (wd_detect_time + wd_poll_time)".format(self.all_detect_time[9],
-                                                                         config_detect_time))
-        pytest_assert(self.all_detect_time[9] < config_detect_time, err_msg)
-
-        if self.timers['pfc_wd_poll_time'] < self.timers['pfc_wd_detect_time']:
-            logger.info("Verify that real detection time is not less than configured")
-            err_msg = ("Real detection time is less than configured: Real detect time: {} "
-                       "Expected: {} (wd_detect_time)".format(self.all_detect_time[9],
-                                                              self.timers['pfc_wd_detect_time']))
-            pytest_assert(self.all_detect_time[9] > self.timers['pfc_wd_detect_time'], err_msg)
-
-        if self.timers['pfc_wd_poll_time'] < self.timers['pfc_wd_restore_time']:
-            logger.info("Verify that real restoration time is not less than configured")
-            err_msg = ("Real restoration time is less than configured: Real restore time: {} "
-                       "Expected: {} (wd_restore_time)".format(self.all_restore_time[9],
-                                                               self.timers['pfc_wd_restore_time']))
-            pytest_assert(self.all_restore_time[9] > self.timers['pfc_wd_restore_time'], err_msg)
-
-        logger.info("Verify that real restoration time is less than configured")
-        config_restore_time = self.timers['pfc_wd_restore_time'] + self.timers['pfc_wd_poll_time']
-        err_msg = ("Real restoration time is greater than configured: Real restore time: {} "
-                   "Expected: {} (wd_restore_time + wd_poll_time)".format(self.all_restore_time[9],
-                                                                          config_restore_time))
-        pytest_assert(self.all_restore_time[9] < config_restore_time, err_msg)
+        logger.info("Verify that real dut detection-restoration time is less than expected value")
+        err_msg = ("Real dut detection-restoration time is greater than configured: Real dut detection-restore time: {}"
+                   " Expected: {}".format(self.all_dut_detect_restore_time[0], self.test_storm_time))
+        pytest_assert(self.all_dut_detect_restore_time[9] < self.test_storm_time, err_msg)
```

Comment:
What's the purpose of "self.test_storm_time", is it the same behavior as the original function?

Reply:
self.test_storm_time represents total time taken for test code to start and stop a storm. Here we want to assert that the time between when DUT detects a storm and when storm is restored (based on syslog timestamps) is less than self.test_storm_time (e.g. if self.test_storm_time is 20 seconds, it would not be valid for the detect/restore time to be 30 seconds).
This is different from the original function, which required the time delta of the storm start and the detection (and likewise, the delta of the storm end and restoration) to be within a fixed limit of 800ms, which is too restrictive.
This PR https://github.com/sonic-net/sonic-mgmt/pull/9416 changed the behaviour for T2 to follow this new behaviour. My PR is effectively doing the same but for all topologies.

Reply:
In the original function, it would verify the real detect/restore time by comparing the timestamp delta between PTF and DUT with the configuration value. 
We have never encountered any errors in this check during our nightly test for T0 and T1, may you share more details about the purpose of this change, have you experienced a high failure rate in this check?